### ISSUE TITLE
Consolidate interface for common behaviors modifying `vtkMRMLSliceNode`

### ISSUE THREAD

**Issue by tbirdso** [Slicer/Slicer#6979]
## Is your feature request related to a problem? Please describe.

I've recently encountered some frustration when dealing with the interface to programmatically adjust slice planes. The "Reformat" module provides a useful grouping of possible controls to adjust slice views, but methods for setting various common slice properties are fragmented among several different, relevant data, logic, and widget classes.

## Describe the solution you'd like

It would be nice to review how common slice node operations are structured and consolidate helper methods into the base `vtkMRMLSliceNode` public interface where it makes sense.

## Additional Details

In general, we have at least four different possible API entry points for updating slice properties:
- `qSlicerReformatModuleWidget`
- `vtkSlicerReformatLogic`
- `vtkMRMLSliceLogic`
- `vtkMRMLSliceNode`

The base `vtkMRMLSliceNode` is where actual slice node properties are stored, but the various other classes encapsulate different levels of helper functionality that is useful for reformatting views. I've investigated each of the four classes above, and the unfortunate reality is that behavior is not neatly divided into widget/logic/data classes.

Referencing the set of slice adjustment behaviors given in the "Reformat" module UI, I've reviewed the interface and internals of the four classes above. I've listed my results below, using "*" to indicate the apparent preferred method for each case.

__Set Slice Visibility__
- \* `vtkMRMLSliceNode` alone exposes methods to toggle slice visibility.

__Set Slice Offset__
- `qSlicerReformatModuleWidget::SetSliceOffset` wraps `vtkMRMLSliceLogic::SetSliceOffset` with start/stop interaction methods. Seems strange to have an external class initiate interaction start/stop events manually.
- `vtkMRMLSliceLogic::SetSliceOffset` simply calls \* `vtkMRMLSliceNode::SetSliceOffset` with a note that the `vtkMRMLSliceLogic` method is available for backwards compatibility

__Set Slice Origin__
- `qSlicerReformatModuleWidget::setWorldPosition` simply calls `vtkSlicerReformatLogic::SetSliceOrigin`
- \* `vtkSlicerReformatLogic::SetSliceOrigin` clamps the origin to the volume bounds and directly updates the `vtkMRMLSliceNode` underlying matrices
- `vtkMRMLSliceLogic` has no public method for this
- `vtkMRMLSliceNode` has `SetSliceOrigin` but produces unexpected behavior different from that in the Reformat module. API methods for jumping slices also produces unexpected behavior with no update in two dimensions.

__Center Slice__
- \* `qSlicerReformatModuleWidget::centerSliceNode` alone exposes a method for centering the slice.

__Reset Slice Orientation to Anatomical Axis__
- \* `vtkMRMLSliceNode` alone exposes a method for resetting the slice orientation to a string enumerating an anatomical plane.

__Set Slice Normal__
- `qSlicerReformatModuleWidget::setSliceNormal` calls `vtkSlicerReformatLogic::setSliceNormal`
- \* `vtkSlicerReformatLogic::SetSliceNormal` normalizes input and directly updates the `vtkMRMLSliceNode` underlying matrices
- `vtkMRMLSliceLogic` and `vtkMRMLSliceNode` do not expose similar methods

__Set Slice Normal To Spatial Axis__
- \* `qSlicerReformatModuleWidget::setSliceNormal` alone defines a method to map an enumerated axis to a normal vector.

## Describe alternatives you've considered

There is a feasible path to evoke each of the common slice adjustment behaviors above. However, the interface is splintered among classes and dealing with slice planes programmatically is unpleasant as a result.

## Additional context

Adding Slicer issue for visibility per encouragement from @jcfr .

**Comment by pieper** [comment]
Yes, moving it to the Legacy category would be fine with me and yes, it can go away.

**Comment by tbirdso** [comment]
Hi all, thanks for the responses here. While interactive slice rotations may present an alternate path for some user interactions in the UI, I'm 1) uncertain whether interactive slice rotations fully encompass the behavior in the Reformat module, and 2) not clear on the proposed path forward for programmatically setting slice properties. For instance, if another Python scripted module wants to center a slice within a volume, it doesn't seem like interactive slice rotations give a clear interface to do that.

In the original investigation above I found four "set" behaviors defined in the Reformat module without a clear parallel in underlying slice node classes:
- `vtkSlicerReformatLogic::SetSliceOrigin`: clamps the origin to the volume bounds and directly updates the vtkMRMLSliceNode underlying matrices
- `qSlicerReformatModuleWidget::centerSliceNode`: center a slice within the volume bounds
- `vtkSlicerReformatLogic::SetSliceNormal`: accepts an arbitrary normal vector, normalizes input and directly updates the vtkMRMLSliceNode underlying matrices
- `qSlicerReformatModuleWidget::setSliceNormal`:  maps an enumerated axis to a normal vector.

Could someone please comment on the proposed alternate path for each of these behaviors? If there is not a duplicate path outside of the Reformat module, could behavior be consolidated (perhaps in `vtkMRMLSliceLogic`) to avoid a regression by moving the Reformat module to Legacy?
